src/BugsnagUnity/Native/Android/NativeInterface.cs

```diff
@@ -454,116 +445,89 @@ private Breadcrumb ConvertToBreadcrumb(IntPtr javaBreadcrumb)
 
       return new Breadcrumb(name, timestamp, typeName, metadata);
     }
 
-    private AndroidJavaObject JavaMapFromDictionary(IDictionary<string, string> src) {
-      var map = new AndroidJavaObject("java.util.HashMap");
-      var CharsetClass = new AndroidJavaClass("java.nio.charset.Charset");
-      var charset = CharsetClass.CallStatic<AndroidJavaObject>("defaultCharset");
+    private AndroidJavaObject MakeJavaMapRef(IDictionary<string, string> src) {
+      AndroidJavaObject map = new AndroidJavaObject("java.util.HashMap");
       if (src != null)
       {
         foreach(var entry in src)
         {
-          map.Call<AndroidJavaObject>("put", MakeJavaString(entry.Key), MakeJavaString(entry.Value));
+          using(AndroidJavaObject key = MakeJavaStringRef(entry.Key))
+          using(AndroidJavaObject value = MakeJavaStringRef(entry.Value))
+          {
+            map.Call<AndroidJavaObject>("put", key, value);
+          }
         }
       }
       return map;
     }
 
-    private AndroidJavaObject MakeJavaString(string input) {
+    private AndroidJavaObject MakeJavaStringRef(string input) {
       if (input == null) {
         return null;
       }
 
       try {
-        var CharsetClass = new AndroidJavaClass("java.nio.charset.Charset");
-        // The default encoding on Android is UTF-8
-        var Charset = CharsetClass.CallStatic<AndroidJavaObject>("defaultCharset");
-        byte[] Bytes = Encoding.UTF8.GetBytes(input);
-        return ConstructJavaString(Bytes, Charset);
+        return new AndroidJavaObject("java.lang.String", input);
       } catch (EncoderFallbackException _) {
         // The input string could not be encoded as UTF-8
         return new AndroidJavaObject("java.lang.String");
       }
     }
 
-    private AndroidJavaObject ConstructJavaString(byte[] Bytes, AndroidJavaObject Charset) {
-      if (Unity2019OrNewer) { // should succeed on Unity 2019.1 and above
-        sbyte[] SBytes = new sbyte[Bytes.Length];
-        Buffer.BlockCopy(Bytes, 0, SBytes, 0, Bytes.Length);
-        return new AndroidJavaObject("java.lang.String", SBytes, Charset);
-      } else { // use legacy API on older versions
-        return new AndroidJavaObject("java.lang.String", Bytes, Charset);
-      }
-    }
-
-    private bool IsUnity2019OrNewer() {
-      var CharsetClass = new AndroidJavaClass("java.nio.charset.Charset");
-      var Charset = CharsetClass.CallStatic<AndroidJavaObject>("defaultCharset");
-
-      try { // should succeed on Unity 2019.1 and above
-        var obj = new AndroidJavaObject("java.lang.String", new sbyte[0], Charset);
-        return true;
-      } catch (System.Exception _) { // use legacy API on older versions
-        return false;
-      }
-    }
-
     private bool CanRunJNI() {
       return CanRunOnBackgroundThread || object.ReferenceEquals(Thread.CurrentThread, MainThread);
     }
 
     private Dictionary<string, object> DictionaryFromJavaMap(IntPtr source) {
```

Comment:
It may be worth investigating whether it's possible to use a combination of [EnsureLocalCapacity](https://docs.unity3d.com/ScriptReference/AndroidJNI.EnsureLocalCapacity.html) and [PushLocalFrame](https://docs.unity3d.com/ScriptReference/AndroidJNI.PushLocalFrame.html) here and in any other methods which require large numbers of local references.